Sources/Spezi/Module/Capabilities/Communication/StorageValueCollector.swift

```diff
@@ -0,0 +1,30 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+
+/// A adopter of this protocol is a property of a ``Component`` that provides mechanisms to retrieve
+/// data provided by other ``Component``s.
+///
+/// Data requested through a Storage Value Collector might be provided through a ``StorageValueProvider``.
+public protocol StorageValueCollector {
```

Comment:
Do we want to explore this in the documentation? If not, we might want to add an underscore before the Type name. Might also apply to some other types that are currently public. Also ensures that they don't show up in autocomplete outside of the target.

Reply:
Probably the use case for extending this from other modules is rather small. We can hide it from documentation for now 👍